bindings/python/scripts/generate.py

```diff
@@ -1,105 +1,404 @@
+                    )
+                else:
+                    self._linelist.append(
+                        f'.def_readwrite("{field["name"]}", &{self.name}::{field["name"]})'
+                    )
+
+        for sub_item in self.members:
+
+            # handle field declarations
+            if sub_item["kind"] == "FIELD_DECL":
+                if sub_item["element_type"] == "ConstantArray":
+                    self._linelist.append(
+                        f'.def_property_readonly("{sub_item["name"]}", []({self.name}& obj) {{return obj.{sub_item["name"]}; }})'  # float[ ' + f'obj.{sub_item["name"]}' + '.size()];} )'
+                    )
+                else:
+                    self._linelist.append(
+                        f'.def_readwrite("{sub_item["name"]}", &{self.name}::{sub_item["name"]})'
+                    )
+
+            # handle class methods
+            elif sub_item["kind"] == "CXX_METHOD":
+                # TODO: Add template args, currently blank
+                if sub_item["name"] not in ("PCL_DEPRECATED"):
+                    self._linelist.append(
+                        f'.def("{sub_item["name"]}", py::overload_cast<>(&{self.name}::{sub_item["name"]}))'
+                    )
+
+    def handle_constructor(self) -> None:
+        """
+        Handles `CursorKind.CONSTRUCTOR`
+
+        - Bind the constructor by developing a parameter list.
+        """
+
+        parameter_type_list = []
+
+        # generate parameter type list
+        for sub_item in self.members:
+            if sub_item["kind"] == "PARM_DECL":
+                if sub_item["element_type"] == "LValueReference":
+                    for sub_sub_item in sub_item["members"]:
+                        if sub_sub_item["kind"] == "TYPE_REF":
+                            # @TODO: Make more robust
+                            type_ref = (
+                                sub_sub_item["name"]
+                                .replace("struct ", "")
+                                .replace("pcl::", "")
+                            )
+                            parameter_type_list.append(f"{type_ref} &")
+                elif sub_item["element_type"] == "Elaborated":
+                    namespace_ref = ""
+                    for sub_sub_item in sub_item["members"]:
+                        if sub_sub_item["kind"] == "NAMESPACE_REF":
+                            namespace_ref += f'{sub_sub_item["name"]}::'
+                        if sub_sub_item["kind"] == "TYPE_REF":
+                            parameter_type_list.append(
+                                f'{namespace_ref}{sub_sub_item["name"]}'
+                            )
+                elif sub_item["element_type"] in ("Float", "Int"):
+                    parameter_type_list.append(f'{sub_item["element_type"].lower()}')
+                else:
+                    parameter_type_list.append(f'{sub_item["element_type"]}')
+        parameter_type_list = ",".join(parameter_type_list)
```

Comment:
This is begging for a refactor, too much nesting. Make a note to extract some `handle_` functions.